bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetAuxiliaryHandler.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.openwebnet.internal.handler;
+
+import static org.openhab.binding.openwebnet.internal.OpenWebNetBindingConstants.CHANNEL_AUX;
+
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.openwebnet.internal.OpenWebNetBindingConstants;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.types.Command;
+import org.openwebnet4j.OpenGateway;
+import org.openwebnet4j.communication.OWNException;
+import org.openwebnet4j.message.Auxiliary;
+import org.openwebnet4j.message.BaseOpenMessage;
+import org.openwebnet4j.message.MalformedFrameException;
+import org.openwebnet4j.message.OpenMessage;
+import org.openwebnet4j.message.UnsupportedFrameException;
+import org.openwebnet4j.message.Where;
+import org.openwebnet4j.message.WhereAuxiliary;
+import org.openwebnet4j.message.Who;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenWebNetAuxiliaryHandler} is responsible for sending Auxiliary (AUX) commands/messages to the bus
+ * It extends the abstract {@link OpenWebNetThingHandler}.
+ *
+ * NOTICE: Support for handling messages from the bus regarding alarm control has to be implemented
+ *
+ * @author Giovanni Fabiani - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class OpenWebNetAuxiliaryHandler extends OpenWebNetThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWebNetAuxiliaryHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = OpenWebNetBindingConstants.AUX_SUPPORTED_THING_TYPES;
+
+    public OpenWebNetAuxiliaryHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        OpenWebNetBridgeHandler h = bridgeHandler;
+        if (h != null && h.isBusGateway()) {
+            OpenGateway gw = h.gateway;
+            if (gw != null && gw.isConnected()) {
+                try {
+                    OpenMessage msg = BaseOpenMessage.parse("*#9##");
+                    send(msg);
+                } catch (MalformedFrameException | UnsupportedFrameException | OWNException e) {
+                    logger.warn("Exception while processing command {}: ", e.getMessage());
```

Comment:
Is this command important ?
No need to change the thing status (to OFFLINE/COMMUNICATION_ERROR for example) in case it fails ?

Reply:
Yes,  that's a good point! But first I would check  how such a modification   would impact  on the feature . I will test it on the plant!

Reply:
I suggest to move the sending of the get status command (`*#9##`) in the requestChannelState method. 
The result will be that anytime a channel needs to be refreshed the generic get status command requesting all AUX states is sent.

Reply:
Ok I made the change as @mvalla suggest. Now I will build and test the feature!

Reply:
@mvalla the  
>move of    sending of the get status command (*#9##) in the requestChannelState method.

cause the  AUX thing remains "stuck" to   UNKNOWN even after  refreshing it. Such a behaviour has come up  during   numerous tests fulfilled on the plant. An evidence of this is crearly shown   in events.log file as well:

> 17:11:13.330 [INFO ] [hab.event.ThingStatusInfoChangedEvent] - Thing 'openwebnet:bus_aux:mybridge:Alarm_activation' changed from INITIALIZING to UNKNOWN 

On the other hand, by leaving the status command (*#9##) in the initialize method the aux thing goes online:
> Thing 'openwebnet:bus_aux:mybridge:Alarm_activation' changed from UNKNOWN: In attesa di aggiornamento di stato... to ONLINE (i.ie. waithing for status update ... to ONLINE)

Reply:
> cause the AUX thing remaining "stuck" to UNKNOWN even after refreshing it

that is because you have not (yet) implemented event message handling, as I already suggested in the comments at the beginning (and in private conversation).

> So the get status command had better stay in the initialize method! Also other developers have overridden the initilialize method for early thing initialization porpouses!

no, because the `initialize()` method should return as fast as possible and not rely on external communication. So sending a message in this method is not a good idea.

My final suggestion is what I suggesting already before: move the sending of the status command (`*#9##`) message in the `requestChannelState()`. 
You also have to complete/implement the `refreshDevice()` method. 
You can just copy similar to what is done in [the Energy handler class](https://github.com/openhab/openhab-addons/blob/5f172c1557ec7f255973f86bbd99e0e20494533b/bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/handler/OpenWebNetEnergyHandler.java#L149) in methods `requestChannelState()` / `refreshDevice()`.

This way at boot and when needed the handler will change Thing status to ONLINE.

Reply:
ok I'll try modify  the implementation as you suggest

Reply:
Ok @mvalla i made the modification,  built the binding and tested it on the plant and ... the thing is online!  Actually I didn't comprehend  how to implement requestChannelState()  correctly to make it process;  the main bulk here was  passing in a new instance of  ChannelUID  and implement it in refreshDevice()!